src/index.js

```diff
@@ -13,49 +13,105 @@ const toGeoJSON = {}
  * */
 
 toGeoJSON.fromCSV = (csv) => {
-    const geojson = { type: 'FeatureCollection', features: [] }
-    let latFieldIndex = null
-    let lonFieldIndex = null
-    let feature, headers
-
-    csv.forEach( (row, i) => {
-        if (i === 0) {
-            headers = row
-
-            // Search a whitelist of lat/longs to try to build a geometry
-
-            headers.forEach((h, i) => {
-                if (['lat', 'latitude', 'latitude_deg', 'y'].indexOf(h.trim().toLowerCase()) > -1) {
-                    latFieldIndex = i + ''
-                } else if (['lon', 'longitude', 'longitude_deg', 'x'].indexOf(h.trim().toLowerCase()) > -1) {
-                    lonFieldIndex = i + ''
-                }
-            })
-        } else {
-            feature = { type: 'Feature', id: i, properties: {}, geometry: null }
-
-            row.forEach((col, j) => {
-                const colNum = col.replace(/,/g, '')
-                feature.properties[headers[j]] = (!isNaN(colNum)) ? parseFloat(colNum) : col
-            })
-
-            // add an object to csv data
-            feature.properties.OBJECTID = i
-
-            if (latFieldIndex && lonFieldIndex) {
-                feature.geometry = {
-                    type: 'Point',
-                    coordinates: [parseFloat(row[parseInt(lonFieldIndex, 10)]), parseFloat(row[parseInt(latFieldIndex, 10)])]
-                }
-            }
-            geojson.features.push(feature)
-        }
+    const geojson = { type: 'FeatureCollection' }
+    const fieldNames = csvFieldNames(csv[0])
+    const features = csv.slice(1)
+    geojson.features = _.map(features, (feat, key) => {
+        const feature = { type: 'Feature', id: key }
+        feature.properties = constructProps(fieldNames, feat)
+        feature.properties.OBJECTID = key
+        feature.geometry = convertCSVGeom(fieldNames, feat)
+        return feature
     })
 
     return geojson ? geojson : null
 }
 
 /**
+ * Parse array of field names and sanitize them
+ *
+ * @param {array} inFields - array of field names
+ * @returns {array} fieldNames - array of sanitized field Names
+ */
+
+function csvFieldNames(inFields) {
+    const fieldNames = []
+     _.each(inFields, (field) => {
+        fieldNames.push(convertFieldName(field))
+    })
+    return fieldNames
+}
+
+/**
+ * Convert CSV geom to geojson
+ *
+ * @param {array} fieldNames - array of field names
+ * @param {array} feature - individual feature
+ * @returns {object} geometry - geometry object
+ */
+
+function convertCSVGeom(fieldNames, feature) {
+    const geometry = {type: 'Point', coordinates: []}
+    _.each(fieldNames, (fieldName, key) => {
+        if (checkForLat(fieldName)) {
+            geometry.coordinates.unshift(parseFloat(feature[key]))
```

Comment:
I think jake is right, it's probably better to use slice and create new variables rather than mutating